bigquery/tests/unit/test_table.py

```diff
@@ -58,6 +58,98 @@ def test_ctor_defaults(self):
         self.assertEqual(table_ref.dataset_id, dataset_ref.dataset_id)
         self.assertEqual(table_ref.table_id, 'table_1')
 
+    def test_to_api_repr(self):
+        from google.cloud.bigquery.dataset import DatasetReference
+        dataset_ref = DatasetReference('project_1', 'dataset_1')
+        table_ref = self._make_one(dataset_ref, 'table_1')
+
+        resource = table_ref.to_api_repr()
+
+        self.assertEqual(
+            resource,
+            {
+                'projectId': 'project_1',
+                'datasetId': 'dataset_1',
+                'tableId': 'table_1',
+            })
+
+    def test_from_api_repr(self):
+        from google.cloud.bigquery.dataset import DatasetReference
+        from google.cloud.bigquery.table import TableReference
+        dataset_ref = DatasetReference('project_1', 'dataset_1')
+        expected = self._make_one(dataset_ref, 'table_1')
+
+        got = TableReference.from_api_repr(
+            {
+                'projectId': 'project_1',
+                'datasetId': 'dataset_1',
+                'tableId': 'table_1',
+            })
+
+        self.assertEqual(expected, got)
+
+    def test___eq___wrong_type(self):
+        from google.cloud.bigquery.dataset import DatasetReference
+        dataset_ref = DatasetReference('project_1', 'dataset_1')
+        table = self._make_one(dataset_ref, 'table_1')
+        other = object()
+        self.assertNotEqual(table, other)
+        self.assertEqual(table, mock.ANY)
+
+    def test___eq___project_mismatch(self):
+        from google.cloud.bigquery.dataset import DatasetReference
```

Comment:
BTW, is there a reason we don't import at the top of the file?

Reply:
Yeah, because we've got some circular references going on, which breaks in Python 2.7.

Reply:
FWIW, https://palladion.com/home/tseaver/obzervationz/2008/unit_testing_notes-20080724 and https://palladion.com/home/tseaver/obzervationz/2009/unit_testing_redux-20090802 have some additional justifications for avoiding imports (particularly of the module/package being tested) at module scope in unit tests.  The point is debated, of course.

Reply:
Thanks for the background!